### PR TITLE
Nitin/code restructure and allure integration

### PR DESCRIPTION
**Pull request by Ni3-Neudesic**
- Created requirment.txt file
- Folder restructuring
- Defined few base level functionalities
- Data driven sample file like json, properties, excel and reading the same
- System independent driver binary download
- Allure reporting
- .git ignored file added
- DocString added

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+pytest
```
Comment (RiteshGitAccount) [inline requirements.txt:1]: Can you please add comment for reason of this imports for better understanding? For all the import mentioned in this file

File: resources/drag_and_drop_helper.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
+(function( $ ) {
```
Comment (RiteshGitAccount) [inline resources/drag_and_drop_helper.js:1]: Add comments on each function for better understanding in future

File: tests/conftest.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    global browser_name
+    browser_name = request.config.getoption("browser_name")
+    wdf = WebDriverFactory()
+    if browser_name == "chrome":
+        driver = wdf.get_chrome_driver()
+    elif browser_name == "firefox":
+        # driver = webdriver.Firefox(executable_path="D:\\FirefoxDriver\\geckodriver.exe")
+        driver = webdriver.Firefox()
+    elif browser_name == "IE":
+        print("IE driver")
```
Comment (RiteshGitAccount) [inline tests/conftest.py:75]: Please add driver for IE also and remove "print" comment line command

File: utilities/BaseClass.py [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+from selenium.webdriver.support import expected_conditions as EC
+import utilities.CustomeLogger as cl
+
+
+class BaseClass:
+    log = cl.custom_logger(logging.DEBUG)
+
+    def __init__(self, driver):
+        self.driver = driver
+
```
Comment (RiteshGitAccount) [inline utilities/BaseClass.py:30]: Please add comment on each function of this file for better understanding in future

**Review by RiteshGitAccount — COMMENTED** [review]
Looks fine to me. Please add comments on Class level(what this class used for) and and function level(what this function used for ). Also if we are using any hardcoded value please make it dynamic

File: tests/conftest.py (hunk 3 above)
Comment (Ni3-Neudesic) [reply tests/conftest.py:75]: Will update it in next PR

File: resources/drag_and_drop_helper.js (hunk 2 above)
Comment (Ni3-Neudesic) [reply resources/drag_and_drop_helper.js:1]: already added

File: utilities/BaseClass.py (hunk 4 above)
Comment (Ni3-Neudesic) [reply utilities/BaseClass.py:30]: already added

File: requirements.txt (hunk 1 above)
Comment (Ni3-Neudesic) [reply requirements.txt:1]: Will update the same in next PR

**Review by RiteshGitAccount — APPROVED** [review]

**Pull request merged**